enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/lucene/SimpleFulltextReader.java

```diff
@@ -85,28 +66,9 @@ public void close()
     }
 
     @Override
-    public FulltextIndexConfiguration getConfigurationDocument() throws IOException
-    {
-        IndexSearcher indexSearcher = getIndexSearcher();
-        TopDocs docs = indexSearcher.search( new TermQuery( FulltextIndexConfiguration.TERM ), 1 );
-        if ( docs.scoreDocs.length < 1 )
-        {
-            return null;
-        }
-        return new FulltextIndexConfiguration( indexSearcher.doc( docs.scoreDocs[0].doc ) );
-    }
-
-    private ScoreEntityIterator innerQuery( String queryString, boolean matchAll )
+    public ScoreEntityIterator query( String queryString )
     {
         MultiFieldQueryParser multiFieldQueryParser = new MultiFieldQueryParser( properties, analyzer );
-        if ( matchAll )
-        {
-            multiFieldQueryParser.setDefaultOperator( QueryParser.Operator.AND );
-        }
-        else
-        {
-            multiFieldQueryParser.setDefaultOperator( QueryParser.Operator.OR );
-        }
         Query query;
         try
         {
```

Comment:
why query error is the same as empty result?

Reply:
Good catch, we need to put something better here.

Reply:
Upon further consideration, I think we should just throw this guy, since in some cases the query might be user-provided. If we are to catch it , we should do so in the bloom surface... Maybe.